spatialdata/_core/_spatialdata.py

```diff
@@ -0,0 +1,262 @@
+    def write(self, file_path: str) -> None:
+        """Write to Zarr file."""
+
+        store = parse_url(file_path, mode="w").store
+        root = zarr.group(store=store)
+
+        # get union of unique ids of all elements
+        elems = set().union(
+            *[
+                set(i)
+                for i in [
+                    self.images,
+                    self.labels,
+                    self.points,
+                    self.polygons,
+                    self.shapes,
+                ]
+            ]
+        )
+
+        for el in elems:
+            elem_group = root.create_group(name=el)
+            if self.images is not None and el in self.images.keys():
+                write_image(
+                    image=self.images[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.labels is not None and el in self.labels.keys():
+                write_labels(
+                    labels=self.labels[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.polygons is not None and el in self.polygons.keys():
+                write_polygons(
+                    polygons=self.polygons[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.shapes is not None and el in self.shapes.keys():
+                write_shapes(
+                    shapes=self.shapes[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+            if self.points is not None and el in self.points.keys():
+                write_points(
+                    points=self.points[el],
+                    group=elem_group,
+                    name=el,
+                    storage_options={"compressor": None},
+                )
+
+        if self.table is not None:
+            write_table(table=self.table, group=root, name="table")
+
+    @property
+    def table(self) -> AnnData:
+        return self._table
+
+    @classmethod
+    def read(cls, file_path: str) -> SpatialData:
+        from spatialdata._io.read import read_zarr
+
+        sdata = read_zarr(file_path)
+        return sdata
+
+    def __repr__(self) -> str:
+        return self._gen_repr()
+
+    def _gen_repr(
+        self,
+    ) -> str:
+        def rreplace(s: str, old: str, new: str, occurrence: int) -> str:
+            li = s.rsplit(old, occurrence)
+            return new.join(li)
+
+        def h(s: str) -> str:
+            return s
+            # return hashlib.md5(repr(s).encode()).hexdigest()
+
+        ##
+        descr = "SpatialData object with:"
+        for attr in ["images", "labels", "points", "polygons", "shapes", "table"]:
```

Comment:
nitpick: this function would likely be cleaner if we had a separate function for generating reprs for each element. We could then generate the repr for each element type and then concatenate them. This would also allow as to move a lot of the details off of the `SpatialData` class into util functions. I think we should strive to make the `SpatialData` class as simple/clean as possible since this will be most people's main contact with the library.

In general, I think long if/else blocks like this are hard to follow and hard to test.

Reply:
Created a issue on this since not high priority, but to be done since it will improve the code quality

Reply:
Thanks!